src/internal/observable/pairs.ts

```diff
@@ -3,6 +3,23 @@ import { Observable } from '../Observable';
 import { SchedulerLike } from '../types';
 import { from } from './from';
 
+/**
+ * @deprecated To be removed in version 8. Use `from(Object.entries(obj))` instead.
+ */
+export function pairs<T>(arr: T[], scheduler?: SchedulerLike): Observable<[string, T]>;
```

Comment:
nitpick: This should probably be `arr: readonly T[]` so that callers can pass read-only arrays.